components/channel/ChannelParameterUnwrap.js

```diff
@@ -0,0 +1,61 @@
+
+import { camelCase, castToTsType, realizeChannelNameWithoutParameters } from '../../utils/index';
+export function unwrap(channelName, channelParameters) {
+  if (Object.keys(channelParameters).length === 0) {
+    return '';
+  }
+  let parameterSplit = '';
+  let prevParameterName = null;
+  parameterSplit = Object.entries(channelParameters).map(([parameterName, _]) => {
+    let toReturn;
+    const parameterCamelCase = camelCase(parameterName);
+    if (prevParameterName) {
+      toReturn = `var ${parameterCamelCase}Split = ${prevParameterName}Split[1].split("${`{${parameterName}}`}");`;
+    } else {
+      toReturn = `var ${parameterCamelCase}Split = unmodifiedChannel.split("${`{${parameterName}}`}");`;
+    }
+    prevParameterName = parameterCamelCase;
+    return toReturn;
+  });
+
+  let splits = '';
+  splits = Object.entries(channelParameters).map(([parameterName, _], index) => {
+    if (index+1 === Object.keys(channelParameters).length) {
+      return `
+			${camelCase(parameterName)}Split[0],
```

Comment:
If you want add indentation at beginning of some texts, you can use helpers from `sdk` like here https://github.com/asyncapi/template-for-generator-templates/blob/main/components/common.js#L53 and pass additional parameter to `unwrap` function like, `intendBy` and it should be a number of spaces/tabs :) You can use it here and in another places in this file, of course :)

Reply:
I just feel like the complexity increases unnecessary, since it does nothing but to "look" good 😄 But yea could be an option... Otherwise do you think I can use eslint to autofix indentation of the output code @magicmatatjahu 🤔 ?

Reply:
Eslint should helps, but I don't know how it will fix it.

Reply:
You can make eslint auto fix errors i.e. indentation, so I dont have to worry about it in each component. But the question is whether it can auto fix it, think I will save it for another issue 😄

Reply:
Ok, make it as followup :)

Reply:
https://github.com/asyncapi/ts-nats-template/issues/20